packages/core/src/Asset/util/metadatumToCip25.ts

```diff
@@ -85,28 +84,67 @@ const mapFile = (metadatum: Metadatum, assetId: Cardano.AssetId, logger: Logger)
 };
 
 /**
- * Also considers asset name encoded in utf8 within metadata valid
+ * Gets the `Map<AssetName, NFTMetadata>` relative to the given policyId from the given
+ * `Map<PolicyId, Map<AssetName, NFTMetadata>>`.
+ *
+ * The policyId in the `policy` Map can be encoded as per CIP-0025 v1 or v2 specifications.
+ *
+ * @param policy The `MetadatumMap` containing the NFT metadata for all the NFT assets
+ * @param policyId The policyId of the assent looking for
+ * @returns The `MetadatumMap` containing the NFT metadata for all the NFT assets with the given policyId
  */
-const getAssetMetadata = (policy: MetadatumMap, asset: Pick<AssetInfo, 'name'>) =>
-  asMetadatumMap(policy.get(asset.name.toString()) || policy.get(Buffer.from(asset.name, 'hex').toString('utf8')));
+const getPolicyMetadata = (policy: Cardano.MetadatumMap, policyId: Cardano.PolicyId) => {
+  const policyIdString = policyId.toString();
+
+  return asMetadatumMap(
+    policy.get(policyIdString) ||
+      (() => {
+        for (const [key, value] of policy.entries()) {
+          if (key instanceof Uint8Array && Buffer.from(key).toString('hex') === policyIdString) return value;
```

Comment:
This is a more reliable check. Same change request for `getAssetMetadata`.
```suggestion
          if (ArrayBuffer.isView(key) && Buffer.from(key).toString('hex') === policyIdString) return value;
```

Reply:
9909f4149 should solve this